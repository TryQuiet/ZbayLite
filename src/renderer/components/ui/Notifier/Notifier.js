import React, { useEffect, useState } from 'react'
import * as R from 'ramda'
import { withSnackbar } from 'notistack'

export const Notifier = ({ notifications, enqueueSnackbar, removeSnackbar }) => {
  const [displayed, setDisplayed] = useState([])
  useEffect(() => {
    notifications.map(
      n => {
        if (displayed.includes(n.key)) {
          return
        }
        enqueueSnackbar(n.message, n.options)
        setDisplayed([...displayed, n.key])
        removeSnackbar(n.key)
      }
    )
  }, [notifications])
  return null
}

export default R.compose(
  React.memo,
  withSnackbar
)(Notifier)
