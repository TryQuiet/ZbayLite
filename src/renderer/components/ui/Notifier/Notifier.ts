import React, { useEffect, useState } from 'react'

interface NotifierProps {
  notifications: Notification[]
  enqueueSnackbar: (message: string, options: any) => void
  removeSnackbar: (key: string) => void
}

interface Notification {
  key: string
  message: string
  options: {
    [s: string]: any
  }
}

export const Notifier: React.FC<NotifierProps> = ({
  notifications,
  enqueueSnackbar,
  removeSnackbar
}) => {
  const [displayed, setDisplayed] = useState([])
  useEffect(() => {
    notifications.map(n => {
      if (displayed.includes(n.key)) {
        return
      }
      enqueueSnackbar(n.message, n.options)
      setDisplayed([...displayed, n.key])
      removeSnackbar(n.key)
    })
  }, [notifications])
  return null
}

export default Notifier
