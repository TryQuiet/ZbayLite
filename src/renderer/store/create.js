import { applyMiddleware, createStore, compose } from 'redux'
import thunk from 'redux-thunk'
import promise from 'redux-promise-middleware'
import createDebounce from 'redux-debounced'

import reducers from './reducers'
import { errorsMiddleware } from './middlewares'

const composer = (enhancers) => {
  if (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ &&
      process.env.NODE_ENV === 'development') {
    return window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__(enhancers)
  }
  return compose(enhancers)
}

export default ({
  initialState
} = {}) => createStore(
  reducers,
  initialState,
  composer(applyMiddleware(...[errorsMiddleware, createDebounce(), thunk, promise()]))
)
