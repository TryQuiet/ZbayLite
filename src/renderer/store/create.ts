import { applyMiddleware, createStore } from 'redux'
import thunk from 'redux-thunk'
import promise from 'redux-promise-middleware'
import createDebounce from 'redux-debounced'
import createSagaMiddleware from 'redux-saga'
import rootSaga from '../sagas/index.saga'
import { composeWithDevTools } from 'redux-devtools-extension'

import reducers from './reducers'
import { errorsMiddleware } from './middlewares'

const sagaMiddleware = createSagaMiddleware()

export default (initialState = {}) => {
  const store = createStore(
    reducers,
    initialState,
    composeWithDevTools(
      applyMiddleware(...[errorsMiddleware, createDebounce(), sagaMiddleware, thunk, promise()])
    )
  )
  sagaMiddleware.run(rootSaga)
  return store
}
