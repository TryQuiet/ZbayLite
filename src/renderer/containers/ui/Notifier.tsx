import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import Notifier from '../../components/ui/Notifier'
import notificationsHandlers from '../../store/handlers/notifications'
import notificationsSelectors from '../../store/selectors/notifications'

export const mapStateToProps = state => ({
  notifications: notificationsSelectors.data(state)
})

export const mapDispatchToProps = dispatch => bindActionCreators({
  removeSnackbar: notificationsHandlers.actions.removeSnackbar
}, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(Notifier)
