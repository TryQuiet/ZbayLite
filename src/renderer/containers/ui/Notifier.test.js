/* eslint import/first: 0 */

import { mapStateToProps, mapDispatchToProps } from './Notifier'

import create from '../../store/create'

describe('Notifier', () => {
  let store = null
  beforeEach(() => {
    jest.clearAllMocks()
    store = create({
      // TODO: 07/05 change to Record
      notifications: [
        {
          message: 'Test message',
          key: 'test-key',
          options: {
            persist: true
          }
        }
      ]
    })
  })

  it('will receive right props', () => {
    const props = mapStateToProps(store.getState())
    expect(props).toMatchSnapshot()
  })

  it('will receive right actions', () => {
    const actions = mapDispatchToProps(x => x)
    expect(actions).toMatchSnapshot()
  })
})
