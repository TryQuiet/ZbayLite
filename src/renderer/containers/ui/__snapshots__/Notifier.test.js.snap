// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifier will receive right actions 1`] = `
Object {
  "removeSnackbar": [Function],
}
`;

exports[`Notifier will receive right props 1`] = `
Object {
  "notifications": Array [
    Object {
      "key": "test-key",
      "message": "Test message",
      "options": Object {
        "persist": true,
      },
    },
  ],
}
`;
